extensions/auth/portability-auth-microsoft/src/main/java/org/dataportabilityproject/auth/microsoft/MicrosoftAuthServiceProvider.java

```diff
@@ -0,0 +1,23 @@
+package org.dataportabilityproject.auth.microsoft;
+
+import org.dataportabilityproject.spi.gateway.auth.AuthDataGenerator;
+import org.dataportabilityproject.spi.gateway.auth.AuthServiceProvider;
+
+/**
+ *
+ */
+public class MicrosoftAuthServiceProvider implements AuthServiceProvider {
```

Comment:
Could you explain a bit how/where the MicrosoftAuthServiceProvider will register itself with the AuthServiceProviderRegistry?

Reply:
Really good question! We need to discuss this as the second part of modularity: how are runtimes assembled with extensions. My understanding is that currently "uber" jars are created but I think this may become difficult to maintain if people want to create and deploy extensions that are not part of the repository. 

There are probably a number of ways we can approach this. For example, using Guice and a bootstrapper, JPMS ServiceLoaders, classpath ServiceLoaders, something else, or a combination of the above. Most of these approaches will involve each extension having some type of bootstrap assembler class/provider that is discovered and invoked. For testing, fixtures would do this work.

To accommodate this, I left the assembly and registration of these extensions for later until we address those issues.  So basically, assembly will be done by some external mechanism TBD. 

Let me know if you have some thoughts on what might work here.

Reply:
Yes currently we have an Uber jar and the ServiceProviderRegistry which registers and loads serviceProviders based on the api settings file. https://github.com/google/data-portability/blob/1944be15d1753da6dc63681f5de7216707ea0b41/portability-core/src/main/java/org/dataportabilityproject/ServiceProviderRegistry.java#L55

For a first version, we can probably do something similar and discuss the dynamic loading requirements later after everything is ported over.

Reply:
Sounds good!